### PR TITLE
Remove any chance of incorrect score handling due to missing entries in `score_process_queue`

### PR DESCRIPTION
**Pull request by peppy**
While working through the import process today to ensure we have everything covered, I realised that the previous fix I made for  ensuring scores are processed was insufficient due to the `null` allowance. Because score submission in web-10 is not done within a transaction, there's a window of time where rows are in `osu_scores_high` but not in `score_process_queue`.

To resolve this, we need to assume two modes of operation:

- Batch processing, when `StartId` is non-null, which will now process from the specified ID up to the last entry in `osu_scores_high` which doesn't have a `score_process_queue` entry (generally two days before current time, due to partition cycling).
- Live mode, when `StartId` is null, which will now process only entries with a `score_process_queue` entry.

I haven't tested this yet, PRing for feedback in the mean time.

### CODE REVIEW

**Review by bdach — CHANGES_REQUESTED** [review]
Not sure I know about enough about the tables / osu-web-10 to provide a meaningful review on that front but ---

File: osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/ImportHighScoresCommand.cs [hunk 1]
```diff
@@ -163,27 +160,45 @@ public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
...
+            ulong maxProcessableId = ulong.MaxValue;
+
+            if (singleRun)
+            {
+                // when doing a single run, we need to make sure not to run into scores which are in the process queue (to avoid
+                // touching them while they are still being written).
+                using (var db = DatabaseAccess.GetConnection())
+                    maxProcessableId = db.QuerySingle<ulong?>("SELECT MIN(score_id) FROM score_process_queue") - 1 ?? ulong.MaxValue;
+
+                where = "WHERE score_id >= @lastId AND score_id <= @";
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/ImportHighScoresCommand.cs:173]: Your guess is as good as mine as to how this happened.

**Review by bdach — APPROVED** [review]
looks better in theory

**Comment by smoogipoo** [comment]
Just so I understand, I basically need to now specify `--start-id=0` every time I need to do processing? i.e. https://github.com/smoogipoo/diffcalc-sheet-generator/blob/170990b323cb421e0a969d14c3a6ee3730858abc/docker/importer/start.sh#L29-L38

**Comment by peppy** [comment]
> Just so I understand, I basically need to now specify `--start-id=0` every time I need to do processing?

Yep, and remove `exit-on-completion`.

**Pull request merged**